ciao-controller/compute.go

```diff
@@ -998,14 +1003,24 @@ func tenantServersAction(w http.ResponseWriter, r *http.Request, context *contro
 	}
 
 	if len(servers.ServerIDs) > 0 {
-		/* TODO Check that instance belongs to the right tenant */
-		for _, instance := range servers.ServerIDs {
-			actionFunc(instance)
+		for _, instanceID := range servers.ServerIDs {
+			// make sure the instance belongs to the tenant
+			instance, err := context.ds.GetInstance(instanceID)
+
+			if err != nil {
+				instanceError := fmt.Sprintf("Instance %s could not be found", instanceID)
```

Comment:
@leoswaldo Could you please make `returnErrorCode()` a variadic function to avoid explicitly building these strings ?
The rest if the code looks fine to me.